tests/unit/pages/fileManager.spec.tsx

```diff
@@ -0,0 +1,425 @@
+        await waitFor(() => {
+          expect(screen.getByText('File already exists')).toBeVisible();
+          fireEvent.click(screen.getByRole('button', { name: /Continue/i }));
+          expect(screen.queryByText('file.txt uploaded successfully.')).not.toBeInTheDocument();
+        });
+      });
+    });
+  });
+});
+describe('File viewer', () => {
+  describe('On load with an existing work number', () => {
+    it('initialises the page', () => {
+      act(() => {
+        require('react-router-dom').useLocation.mockImplementation(() => {
+          return {
+            pathname: () => './file_viewer',
+            search: 'workNumber=SGP1008'
+          };
+        });
+        renderFileManagerComponent(false);
+      });
+      expect(screen.queryByText('Upload file')).not.toBeInTheDocument();
+      expect(screen.queryByText('Files')).toBeVisible();
+    });
+  });
+  describe('On load for a non-existing SGP', () => {
+    it('should display warning', () => {
+      act(() => {
+        require('react-router-dom').useLocation.mockImplementation(() => {
+          return {
+            pathname: () => './file_viewer',
+            search: 'workNumber=Blah'
+          };
+        });
+        renderFileManagerComponent(false);
+      });
+      expect(screen.queryByText('SGP Number(s) does not exist.')).toBeVisible();
+      expect(screen.queryByText('Files')).not.toBeInTheDocument();
+    });
+  });
+});
+describe('Active checkbox', () => {
+  beforeEach(() => {
+    require('react-router-dom').useLocation.mockImplementation(() => {
+      return {
+        pathname: () => './file_manager',
+        search: 'workNumber=SGP1008'
+      };
+    });
+    require('react-router-dom').useLoaderData = jest.fn().mockReturnValue([
+      {
+        workNumber: 'SGP1008',
+        workRequester: 'user1',
+        project: 'Project 1',
+        status: WorkStatus.Active
+      }
+    ]);
+    renderFileManagerComponent();
+  });
+  describe('when active checkbox is selected', () => {
+    it('should display a single option', async () => {
+      await workNumberNumOptionsShouldBe(2);
+    });
+  });
+  describe('when active checkbox is unselected', () => {
+    act(() => {
+      uncheck();
+    });
+    it('should display more options', () => {
+      workNumberNumOptionsShouldBeMoreThan(2);
+    });
+  });
+});
+const renderFileManagerComponent = (showUpload = true) => {
+  return render(
+    <BrowserRouter>
+      <FileManager showUpload={showUpload} />
+    </BrowserRouter>
+  );
+};
+
+const selectFile = async () => {
+  const file = new File(['sample'], 'file.txt', {
+    type: 'text/plain'
+  });
+  await waitFor(() => {
+    fireEvent.change(screen.getByTestId('file-input'), { target: { files: [file] } });
+  });
+};
+
+const uploadFile = async () => {
+  await selectFile();
+  expect(screen.getByTestId('upload-btn')).toBeEnabled();
+  await userEvent.click(screen.getByTestId('upload-btn'));
+};
```

Comment:
The test case  when visting the page as an end user is missing. It can either be added here or as a cypress e2e test in fileManger.cy.ts